Document guidelines for every field (#255)
axelboc/anki-ultimate-geography: CONTRIBUTING.md

#### _Country info_ field

To help with memorisation and provide context while learning, this field may contain:

- governance information - e.g. _Overseas territory of the United Kingdom_ (Cayman Islands)
- statehood information - e.g. _Independent state claimed by Moldova_ (Transnistria)
- alternative and former country names - e.g. _Also known as Timor-Leste_ (East Timor)
- in rare cases only, general knowledge information - see Melanasia for an example

The content of this field should be concise and consistent across notes. It may differ between languages, notably when dealing with alternative names.

#### _Capital_ field

Unless otherwise stated in [`TRANSLATION_SOURCES.md`](https://github.com/axelboc/anki-ultimate-geography/blob/master/TRANSLATION_SOURCES.md), we use the capital given in the [infobox](https://en.wikipedia.org/wiki/Template:Infobox_country#Examples) of the Wikipedia article for the country, in the language of the given deck.

> If a capital has alternative names, we take the **first listed name** in the Wikipedia article for the capital, in the language of the given deck. This corresponds typically to the title of the article and to the name used in the infobox of the country's Wikipedia article. Alternative names may be mentioned in the _Capital info_ field, when relevant.

Reviewer: > If a capital has alternative names, we take the **first listed name** in the Wikipedia article for the capital

That's not what I used for [Laayoune](https://en.wikipedia.org/wiki/Laayoune) — there the first listed name is Laâyoune, but the article title and the name listed in the SADR infobox is just Laayoune. (I assume that you mean the first "bolded" name, not the name in the header?) Should I change it?

Author: That's the rule that was already specified for country names, but I didn't think the page title could mismatch the first listed name... This seems like an oversight on Wikipedia's side, TBH. In the talk page, they discuss that `Laayoune` is more commonly used than `Laâyoune`, so it is indeed two valid and distinct spelling that should both appear in the introduction, IMO.

How about I just update the guideline to mention using the article title?

Author: Okay, I've [updated the guidelines](https://github.com/axelboc/anki-ultimate-geography/pull/255/commits/cfa011ed16c3844711c2933d5cbe2bc509779eca):

- For countries, we use the titles and we remove any disambiguation parenthetical, unless disambiguation is needed in the deck.
- For capitals, we use the titles and we remove any disambiguation parenthetical -- the hint field is there for that.

Reviewer: I think that also works!

If multiple capitals are listed in a country's infobox, the following guidelines apply:

- If the first capital is followed by a qualifier such as "official", "constitutional", "de jure", "claimed" or "political", it must be used alone in the _Capital_ field. The _Capital info_ field must then be used to detail the status and/or role of every capital - e.g. _While Dodoma is the official capital, Dar es Salaam is the de facto seat of government._
- If government branches such as "executive" or "legislative" are the only qualifiers used, then the capitals must all be listed in the _Capital_ field, separated by comas - e.g. _Pretoria, Cape Town, Bloemfontein_ (South Africa). The _Capital info_ field must again be used to detail the role of every capital.

> - If no qualifiers are provided at all, then the capitals must all be listed in the _Capital_ field, separated by comas - e.g. _Santa Cruz de Tenerife, Las Palmas_ (Canary Islands). A concise explanation should then be provided in the _Capital info_ field.

Reviewer: s/comas/commas/ (twice)

> - If government branches such as "executive" or "legislative" are the only qualifiers used, then the capitals must all be listed in the _Capital_ field, separated by comas - e.g. _Pretoria, Cape Town, Bloemfontein_ (South Africa). The _Capital info_ field must again be used to detail the role of every capital.
> - If no qualifiers are provided at all, then the capitals must all be listed in the _Capital_ field, separated by comas - e.g. _Santa Cruz de Tenerife, Las Palmas_ (Canary Islands). A concise explanation should then be provided in the _Capital info_ field.

Reviewer: ```suggestion
- If government branches such as "executive" or "legislative" are the only qualifiers used, then the capitals must all be listed in the _Capital_ field, separated by commas - e.g. _Pretoria, Cape Town, Bloemfontein_ (South Africa). The _Capital info_ field must again be used to detail the role of every capital.
- If no qualifiers are provided at all, then the capitals must all be listed in the _Capital_ field, separated by commas - e.g. _Santa Cruz de Tenerife, Las Palmas_ (Canary Islands). A concise explanation should then be provided in the _Capital info_ field.
```

#### _Capital info_ field

As explained in the previous section, this field is typically used for countries with multiple capitals, to clarify the role and/or status of each capital, or to explain succinctly why the country has multiple capitals.

#### _Capital hint_ field

This field is used in notes that share:

- the same exact capital - e.g. _London_ (United Kingdom and England);
- capitals with very similar names - e.g. _Georgetown_ and _George Town_.

The hints appear on _Capital - Country_ cards to avoid confusion or random guesses. They should

> convey as little information as possible to not giveaway the answers.

Reviewer: ```suggestion
convey as little information as possible to not give away the answers.
```

I think that "give away" as a verb is two words, and [Wiktionary agrees](https://en.wiktionary.org/wiki/giveaway).

- SVG flags larger than 50 kB must be exported to PNG (still with a height of 250 px) and optimised with a tool like [PNGGauntlet](https://pnggauntlet.com/).

If the name of a country appears clearly on a flag, a second version of that flag may be provided, with the name of the country blurred out. The name should be blurred using [Inkscape](https://inkscape.org/)'s Gaussian blur effect as explained in #247. The blurred flag must be named `ug-flag-<country_name>-blur.<svg|png>` and placed in the `media` folder. A second HTML element must then be added to the _Flag_ field _before_ the existing HTML element. This allows the blurred flag to appear on the front of the country's _Flag - Country_ card.

#### _Flag similarity_ field

This field is used in the _Flag - Country_ template.

In Anki, when you keep on confusing two flags, you can't put them side by side to learn their visual differences. The _Flag similarity_ field works around this limitation by providing a concise description of the differences a flag has with another. It allows users to more easily learn to distinguish pairs of similar flags and perhaps come up with mnemonics to remember their respective countries.

> A note's _Flag similarity_ field contains a list of countries, each followed by a list of differences. For instance, the flag similarities for Iceland are written as follows: _Norway (red background, blue cross), Faroe Islands (white background, red and blue cross)_. The list of differences must be **precise**, **clear** and **concise**. Advanced [vexillological](https://en.wikipedia.org/wiki/Vexillology) terms should be avoided.

Reviewer 2: ```suggestion
A note's _Flag similarity_ field contains a list of countries, each followed by a list of differences. For instance, the flag similarities for Iceland are written as follows: _Norway (red background, blue cross), Faroe Islands (white background, red and blue cross)_. The list of differences must be **precise**, **clear**, and **concise**. Advanced [vexillological](https://en.wikipedia.org/wiki/Vexillology) terms should be avoided.
```

Oxford comma to fully separate each item in the list.
![image](https://user-images.githubusercontent.com/1471350/78467692-79ce4f00-770f-11ea-9768-b24ea98aaeba.png)

  - subtly different colours - i.e. [ΔE](https://github.com/axelboc/anki-ultimate-geography/issues/50#issuecomment-525902404) < 30
  - subtly different geometry

Two flags are then eligible for _Flag similarity_ information when they respect the two rules below:

- Their differences all fit in the above classification. For instance, France and Italy are not eligible because their left bands are respectively blue and green, and "different colours in same position" does not appear in the classification.
- Their classified differences form one the following combinations:
  - `1C 0M 0m {0+}n` = one critical difference and any number of negligible differences
  - `0C 1M {0-2}m {0+}n` = one major difference, up to two minor differences and any number of negligible differences
  - `0C 0M {0-3}m {0+}n` = up to three minor differences and any number of negligible differences

> Critical, major and minor differences should be listed in the _Flag similarity_ field. Negligible differences should be listed only when relevant, notably when two flags share nothing but negligible differences.

Reviewer 2: > Critical, major and minor

Oxford comma

Author: Hmm this one hurts the flow of the sentence in my opinion. Oxford comma is good but doesn't have to be a hard rule ;)
